src/control/server/ctl_network_rpc.go

```diff
@@ -49,7 +48,16 @@ func (c *ControlService) NetworkScan(ctx context.Context, req *ctlpb.NetworkScan
 		excludeMap[iface] = struct{}{}
 	}
 
-	results, err := netdetect.ScanFabric(req.GetProvider())
+	provider := req.GetProvider()
+	switch provider {
+	case "":
+		provider = c.srvCfg.Fabric.Provider
+	case "all":
+		provider = ""
+	default:
+	}
```

Comment:
See the suggestion in daos_server for starting with the default value.

Reply:
Done.